### ISSUE TITLE
OpenLineage json schema has invalid relative references of facets

### ISSUE THREAD

**Issue by fm100** [OpenLineage/OpenLineage#1440]
OpenLineage spec defines its [JSON schema](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json), and it has many relative references for facets:
* https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json#L83-L86
* https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json#L122-L126
* https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json#L158
* https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json#L188
* https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json#L223-L231

Because they are relative references, by the JSON schema standard, the reference is translated to absolute URI by combining with the location specified in `$id`. For example, if the `$id` of the main doc is `https://openlineage.io/spec/1-0-5/OpenLineage.json` and the relative reference `"$ref": "facets/ParentRunFacet.json"` the resulting absolute reference becomes `https://openlineage.io/spec/1-0-5/ParentRunFacet.json`. However, the actual ID of `ParentRunFacet` is [`https://openlineage.io/spec/facets/1-0-0/ParentRunFacet.json`](https://github.com/OpenLineage/OpenLineage/blob/main/spec/facets/ParentRunFacet.json#L3) and translated absolute reference is invalid.

Tools like JSON schema validators recursively follow the schema def and compile it, but this blocks the use of many available tools.

**Comment by denimalpaca** [comment]
@fm100 is this closed wrt the above merged PR?

**Comment by fm100** [comment]
> @fm100 is this closed wrt the above merged PR?

No, the PR is related with the issue, but it made a workaround to solve it. The issue still exists in spec.

**Comment by pawel-big-lebowski** [comment]
My understanding of the problem is: 
 * At the beginning of the project all the facets definitions were within single schema. 
 * This lead to problems: each update of a single facet's definition causes version bump of a while Openlineage. 
 * At some point, it got changed according to the [proposal](https://github.com/OpenLineage/OpenLineage/issues/153)
 * The side effect of this is that each facet and Openlineage is versioned separately. Separate versioning causes relative links failures, bcz for latest schema version `https://openlineage.io/spec/1-0-5/OpenLineage.json`, we are using column lineage facet `https://openlineage.io/spec/facets/1-0-1/ColumnLineageDatasetFacet.json`. The schema within `OpenLineage.json` has no idea on latest versions of facets like `ColumnLineageDatasetFacet.json`. Even harder, if it knew that, then upgrade of each facet would cause main schema upgrade, which we would like to avoid.

Perhaps we could introduce latest version for facets like : `https://openlineage.io/spec/facets/LATEST/ColumnLineageDatasetFacet.json` 

@mobuchowski wdyt?

**Comment by mobuchowski** [comment]
I would like to explore idea with removing references to actual facets from main openlineage.yml schema file and how would that work with general generators.
